### DISCUSSION TITLE
Use of Ring cloud servers

### DISCUSSION THREAD

**Discussion by FransOv** [tsightler/ring-mqtt#392] [General Discussion]
Thanks for this project. With the documentation provided I could manually install ring-mqtt with minimal hiccups. The only problem was that it was not mentioned that you had to set the rsp executable as executable. Integration with my Home Assistant installation went flawlessly. 
What I would like to know but what I could not find the answer to is does the ring api connect to the Ring cloud servers or to the Ring Alarm Base station. I.e. can I disconnect the base station  from the internet and will ring-mqtt then still function.

**Comment by tsightler** [comment]
Thanks for the feedback.  When you say "rsp" executable do you mean rtsp-simple-server?

The ring-client-api uses the same Ring cloud API as the Ring app, so no, you cannot disconnect the base station from the internet.  For better or worse, Ring devices are cloud based devices, they are mostly useless bricks when disconnected from the cloud. 

That being said, realistically, if you don't want to connect to the cloud, then what use is the base station at all?  It's basically just a Z-wave hub at that point, so you can instead just throw it away and pair the Ring sensors with any generic Z-wave hub/USB stick that you can control locally, or just buy much cheaper Z-wave sensors and build your own alarm with HA or other tools.

**Comment by FransOv** [comment]
Yes, I did mean the rtsp-simple-server, the t went missing.
I suspected that the Ring cloud was still used, but hoped that it was not. The option to control the sensor and other devices via Z-wave I'll keep in reserve. Interfacing the sensors will be rather straightforward, the  keypad maybe not that easy.

**Comment by tsightler** [comment]
Perhaps not super easy, but completely possible as some have already done it:  https://github.com/ImSorryButWho/HomeAssistantNotes/blob/6f05b3f90a979f7118e7c688ce2d01b878227328/RingKeypadV2.md

There are also some pretty nice Zigbee keypads that a lot of people use.  Personally, if I just wanted a local alarm I'd probably purchase one of these and use the Alarmo integration in Home Assistant, it's pretty good and you can use any sensors you want.

**Comment by FransOv** [comment]
Thank you for the information. Looks promising, I'll check into it.
